content/spin/v2/managing-plugins.md

```diff
@@ -116,27 +117,77 @@ You can uninstall plugins using `spin plugins uninstall` with the plugin name:
 $ spin plugins uninstall befunge2wasm
 ```
 
+## Refreshing the Catalogue
+
+The first time you install a plugin from the catalogue, Spin creates a local cache of the catalogue. It continues to use this local cache for future install, list and upgrade commands; this is similar to OS package managers such as `apt`, and avoids rate limiting on the catalogue. However, this means that in order to see new catalogue entries - new plugins or new versions - you must first update the cache. 
+
+To update your local cache of the catalogue, run `spin plugins update`.
+
 ## Upgrading Plugins
 
-To upgrade a plugin to the latest version, run `spin plugins upgrade`. This has the same options as `spin plugins install`, according to whether the plugin comes from the catalogue, a URL, or a file:
+To upgrade a plugin to the latest version, first run `spin plugins upgrade` (to refresh the catalogue). 
+
+The `spin plugins upgrade` command has the same options as the `spin plugins install` command (according to whether the plugin comes from the catalogue, a URL, or a file). For more information, see the plugins section of the [Spin CLI Reference documentation](cli-reference#spin-plugins). 
+
+> The `upgrade` command uses your local cache of the catalogue. This might not include recently added plugins or versions. So always remember to run `spin plugins update` to refresh your local cache of the catalogue before performing the `spin plugins upgrade` command.
+
+The following example shows how to upgrade one plugin at a time (i.e. the `js2wasm` plugin):
 
 <!-- @selectiveCpy -->
 
 ```bash
+$ spin plugins update
 $ spin plugins upgrade js2wasm
+```
+
+The following example shows how to upgrade all installed plugins at once:
```

Comment:
This only installs plugins from the registry - do we need to express that explicitly?

Reply:
Good point, will add that for clarity.